scheduler/test/cook/test/scheduler/scheduler.clj

```diff
@@ -1493,10 +1591,12 @@
         (is (= (:normal expected-pool->pending-jobs)
                (:normal (sched/remove-matched-jobs-from-pending-jobs pool->pending-jobs (:normal pool->matched-job-uuids) :normal))))))))
 
-(deftest test-handle-resource-offers
+(defn test-handle-resource-offers
+  "Helper function for test-handle-resource offers"
+  [launched-job-names-atom launched-offer-ids-atom offers-list test-user uri]
   (setup)
-  (let [uri "datomic:mem://test-handle-resource-offers"
-        conn (restore-fresh-database! uri)test-user (System/getProperty "user.name")
+  (let [; Mesos needs to know which offers are being used to launch tasks with, whereas kubernetes has no concept of offers
+        is-mesos-compute-cluster? (when launched-offer-ids-atom true)
```

Comment:
Following up with our phone call...

Thanks for making the refactor. It makes things a lot easier to understand. In the code, you have a lot of clauses that look like:

          (when is-mesos-compute-cluster?
            (is (= 4 (count @launched-job-ids-atom)))

So, this clause is disabled for k8s compute clusters. I think we should fix this. We need to do it his way because this code block is mocking out at the mesos driver level (with the (reify msched/SchedulerDriver). Instead, I think we should refactor this code to be mocking out at the compute cluster level, then this code is agnostic to mesos or GKE... 

It *should* be agnostic to mesos or GKE. That way we may avoid some of the other duplication of mesos vs GKE lower down.

We're already doing some of this mocking, look at line 1691:

(with-redefs [kcc/launch-task! (fn [_ task-metadata]

We can determine the number of matched offers for GKE and Mesos. We group matches when we send them to launch-tasks into a list of machines on which we have a list the jobs to run on that machine. (machine = offer)

Reply:
Made higher level refactor at `sched/launch-matched-tasks!` and used reify for compute cluster. There are now no mesos and k8s compute clusters. GPU tests are still unique but mocking is at compute cluster level now. 

As a result, removed one of the tests that was testing`rate-limit`, but this shouldn't be an issue because there are integration tests that test that function thoroughly.

Reply:
Mocked at compute cluster level so this is now good.